src/macros.rs

```diff
@@ -1241,11 +1266,14 @@ impl MacroBranch {
         context: &RewriteContext<'_>,
         shape: Shape,
         multi_branch_style: bool,
-    ) -> Option<String> {
+    ) -> RewriteResult {
         // Only attempt to format function-like macros.
         if self.args_paren_kind != Delimiter::Parenthesis {
             // FIXME(#1539): implement for non-sugared macros.
-            return None;
+            return Err(RewriteError::MacroFailure {
+                macro_error_kind: MacroErrorKind::Unknown,
+                span: self.span,
+            });
         }
```

Comment:
I would like to ask if it is better to add additional `MacroErrorKind` to express this error

Reply:
`MacroErrorKind::Unknown` seems fine to me in this case